lang/java/reef-bridge-proto-java/src/main/java/org/apache/reef/bridge/driver/service/grpc/GRPCDriverService.java

```diff
@@ -0,0 +1,1084 @@
+      this.allocatedEvaluatorMap.remove(eval.getId());
+      this.clientStub.failedEvaluatorHandler(
+          EvaluatorInfo.newBuilder().setEvaluatorId(eval.getId()).build());
+    }
+  }
+
+  @Override
+  public void activeContextHandler(final ActiveContext context) {
+    synchronized (this) {
+      this.activeContextMap.put(context.getId(), context);
+      this.clientStub.activeContextHandler(
+          ContextInfo.newBuilder()
+              .setContextId(context.getId())
+              .setEvaluatorId(context.getEvaluatorId())
+              .setParentId(
+                  context.getParentId().isPresent() ?
+                      context.getParentId().get() : "")
+              .setEvaluatorDescriptorInfo(toEvaluatorDescriptorInfo(
+                  context.getEvaluatorDescriptor()))
+              .build());
+    }
+  }
+
+  @Override
+  public void closedContextHandler(final ClosedContext context) {
+    synchronized (this) {
+      this.activeContextMap.remove(context.getId());
+      this.clientStub.closedContextHandler(
+          ContextInfo.newBuilder()
+              .setContextId(context.getId())
+              .setEvaluatorId(context.getEvaluatorId())
+              .setParentId(context.getParentContext().getId())
+              .setEvaluatorDescriptorInfo(toEvaluatorDescriptorInfo(
+                  context.getEvaluatorDescriptor()))
+              .build());
+    }
+  }
+
+  @Override
+  public void failedContextHandler(final FailedContext context) {
+    synchronized (this) {
+      final ContextInfo.Builder contextInfoBuilder =
+          ContextInfo.newBuilder()
+              .setContextId(context.getId())
+              .setEvaluatorId(context.getEvaluatorId())
+              .setParentId(
+                  context.getParentContext().isPresent() ?
+                      context.getParentContext().get().getId() : "")
+              .setEvaluatorDescriptorInfo(toEvaluatorDescriptorInfo(
+                  context.getEvaluatorDescriptor()));
+      if (context.getReason().isPresent()) {
+        final Throwable reason = context.getReason().get();
+        contextInfoBuilder.setException(ExceptionInfo.newBuilder()
+            .setName(reason.toString())
+            .setMessage(context.getMessage() != null ? context.getMessage() : "")
+            .setData(ByteString.copyFrom(exceptionCodec.toBytes(reason)))
+            .build());
+      } else if (context.getData().isPresent()) {
+        contextInfoBuilder.setException(ExceptionInfo.newBuilder()
+            .setName(context.toString())
+            .setMessage(context.getDescription().isPresent() ?
+                context.getDescription().get() :
+                context.getMessage() != null ? context.getMessage() : "")
+            .setData(ByteString.copyFrom(context.getData().get()))
+            .build());
+      } else {
+        final Throwable reason = context.asError();
+        contextInfoBuilder.setException(ExceptionInfo.newBuilder()
```

Comment:
same. we have to grep the code for all places where we still use `ExceptionInfo.newBuilder()` instead of `.createExceptionInfo()`